src/Features/Core/SolutionCrawler/WorkCoordinator.AsyncWorkItemQueue.cs

```diff
@@ -99,6 +99,15 @@ public bool AddOrReplace(WorkItem item)
                     }
                 }
 
+                public void RequestCancellationOnRunningTasks()
+                {
+                    lock (_gate)
+                    {
+                        // request to cancel all running works
+                        _cancellationMap.Do(p => p.Value.Cancel());
```

Comment:
ya, I thought about that, but decide to go just simple since when running tasks are cancelled, it should correctly clear the cancel map itself. and calling cancel twice on same cancellation source should be fine.